# 1. Guppy Basecaller

```plaintext
Guppy Basecaller is a software tool to convert the raw electrical signal data from nanopore sequencing into DNA sequences. 
Here's an example of a commonly used command with Guppy Basecaller:

guppy_basecaller -i /input/directory -s /output/directory --config configuration.cfg

In this command:

- -i /input/directory specifies the input directory where the raw data files (.fast5 files) are located.
- -s /output/directory specifies the output directory where the basecalled reads will be written.
- --config configuration.cfg is used to specify a configuration file that contains the basecalling model and other settings.

This command will basecall the raw data in the specified input directory using the settings from the configuration file, and write the output to the specified output directory.
```


# 2. Porechop

```plaintext
Porechop is a tool developed for Oxford Nanopore sequencing data. It is used for finding and removing adapters from Oxford Nanopore reads. Adapters on the ends of reads are trimmed off, and when a read has an adapter in its middle, it is treated as chimeric and chopped into separate reads.

Here's an example of a commonly used command with Porechop:

porechop -i input.fastq -o output.fastq

In this command:

- -i input.fastq specifies the input file (in FASTQ format) that you want to trim adapters from.
- -o output.fastq specifies the output file where the trimmed reads will be written.

This command reads the input FASTQ file, trims adapters from the reads, and writes the resulting trimmed reads to the output file.
```


# 3. NanoFilt

```plaintext
NanoFilt is a simple tool to filter Oxford Nanopore sequencing data. It reads in a FASTQ file (or stdin), filters reads based on a minimum quality and/or a minimum length, and writes out the filtered reads to stdout.

Here's an example of a commonly used command with NanoFilt:

gunzip -c input.fastq.gz | NanoFilt -q 10 -l 500 | gzip > output.fastq.gz

In this command:

- gunzip -c input.fastq.gz is used to decompress the input FASTQ file.
- NanoFilt -q 9 -l 500 filters reads based on a minimum quality of 9 and a minimum length of 100.
- gzip > output.fastq.gz compresses the filtered reads and writes them to the output file.

This command decompresses the input FASTQ file, filters the reads based on the specified minimum quality and length, and writes the filtered reads to the output file in compressed format.
```


# 4. Flye

```plaintext
Flye is a de novo assembler for single-molecule sequencing reads, such as those produced by PacBio and Oxford Nanopore Technologies. It is designed for a wide range of datasets, from small bacterial projects to large mammalian genomes. The Flye assembler provides accurate, fast, and scalable solutions to assembly problems.

In the context of metagenomic projects, Flye can be run in 'meta' mode. Here's an example of a commonly used command with Flye in 'meta' mode:

flye --meta --nano-raw input.fastq --out-dir output_directory 

In this command:

- --meta indicates that Flye should be run in 'meta' mode for metagenomic data.
- --nano-raw input.fastq specifies the input file containing the reads. Depending on the source of the sequencing data change to nano-hq or nano-corr
- --out-dir output_directory specifies the directory where the assembly result will be written.

This command runs Flye in 'meta' mode on the input reads, assembles the reads, and writes the assembly result to the specified output directory.
```

# 5. Minimap2

```plaintext
Minimap2 is a versatile sequence alignment program that aligns DNA or mRNA sequences against a large reference database. 

Here's an example of a commonly used command with Minimap2:

minimap2 -ax map-ont /path/to/flye/output/assembly.fasta /path/to/nanofilt/output/reads.fastq | samtools sort -o /path/to/minimap2/output/reads.sorted.bam

In this command:

- -ax map-ont tells minimap2 that the input are Oxford Nanopore reads.
- /path/to/flye/output/assembly.fasta is the file path to the assembled sequences.
- /path/to/nanofilt/output/reads.fastq is the file path to the nanopore reads.
- samtools sort -o /path/to/minimap2/output/reads.sorted.bam sorts the output and specifies the output file path.

This command runs minimap2 and realigns the initial reads to the assemblye assembled by flye.
```

# 6. Racon
```plaintext
Racon is a consensus module to correct raw contigs generated by rapid assembly methods which do not include a consensus step, such as Flye.

Here's an example of a commonly used command with Racon:

racon /path/to/nanofilt/output/reads.fastq /path/to/minimap2/output/reads.sam /path/to/flye/output/assembly.fasta > /path/to/racon/output/assembly.polished.fasta

In this command:

- /path/to/nanofilt/output/reads.fastq is the file path to the nanopore reads.
- /path/to/minimap2/output/reads.sam is the file path to the sorted alignment file.
- /path/to/flye/output/assembly.fasta is the file path to the assembled sequences.
- /path/to/racon/output/assembly.polished.fasta specifies the output file path.
```

# 7. Vamb
```plaintext
Vamb is a tool for metagenomic binning of assembled contigs. It uses variational autoencoders to cluster contigs into metagenome-assembled genomes (MAGs).

Here's an example of a commonly used command with VAMB:

vamb --outdir /path/to/vamb/output --fasta /path/to/flye/output/assembly.fasta --bamfiles /path/to/minimap2/output/reads.sorted.bam
```

# 8. MaxBin2
```plaintext
MaxBin2 is a tool used for binning assembled metagenomic sequences based on an Expectation-Maximization algorithm. Here is a commonly used command:

maxbin2 -input /path/to/assembly.fasta -out /path/to/output -abund /path/to/abundance.txt
```

# 9. MetaBat2
```plaintext
MetaBAT2 is an efficient tool for accurately reconstructing single genomes from complex microbial communities. Here is a typical command for using MetaBAT2:

metabat2 -i /path/to/assembly.fasta -a /path/to/depth.txt -o /path/to/output_bins
```

# 10. Concoct
```plaintext
CONCOCT is a new algorithm that combines sequence composition and coverage across multiple samples to automatically bin metagenomic contigs into genomic bins. Here is how to use CONCOCT:

cut_up_fasta.py /path/to/assembly.fasta -c 10000 -o 0 --merge_last -b contigs_10K.bed > contigs_10K.fa

concoct_coverage_table.py contigs_10K.bed /path/to/mapped/reads_sorted.bam > coverage_table.tsv

concoct --composition_file contigs_10K.fa --coverage_file coverage_table.tsv -b concoct_output/
```

# 11. DAS_Tool
```plaintext
DAS_Tool is a precision-weighted method for automated metagenome de novo assembly structure optimization. Here is an example command to run DAS_Tool:

DAS_Tool -i /path/to/metabat_output.tsv,/path/to/maxbin_output.tsv,/path/to/concoct_output.tsv -l metabat,maxbin,concoct -c /path/to/assembly.fasta -o /path/to/dastool_output
```

# 12. Prodigal
```plaintext
Prodigal (Prokaryotic Dynamic Programming Genefinding Algorithm) is a microbial (bacterial and archaeal) gene finding program.

prodigal -i /path/to/assembly.fasta -a /path/to/proteins.faa -o /path/to/prodigal_output.gbk -p meta
```

# 13. EggNOG-mapper
```plaintext
EggNOG-mapper is a tool for fast functional annotation of novel sequences. It uses precomputed evolutionary models to perform fast orthology assignments.

emapper.py -i /path/to/proteins.faa -o /path/to/eggnog_output -m diamond --cpu 8 --data_dir /path/to/eggnog_data/
```

# 14. AMRFinderPlus
```plaintext
AMRFinderPlus is a tool that identifies antimicrobial resistance genes in protein or nucleotide data.

amrfinder -p /path/to/proteins.faa -o /path/to/amrfinder_output.txt
```

# 15. Quast
```plaintext
QUAST stands for QUality ASsessment Tool. It evaluates genome assemblies by computing various metrics.

quast.py -o /path/to/quast_output/ /path/to/assembly.fasta
```

# 16. CheckM
```plaintext
CheckM provides a set of tools for assessing the quality of genomes recovered from isolates, single cells, or metagenomes. 

checkm lineage_wf /path/to/bins/ /path/to/checkm_output/
```

# 17. CoverM
```plaintext
CoverM is a tool designed to calculate, normalize and visualize the coverage of microbial genomes. 

coverm genome -b /path/to/aligned.bam -r /path/to/assembly.fasta --output-format sparse > /path/to/coverage.tsv
```
