# Air Monitoring by Nanopore Sequencing

## Detailed Overview

This project utilizes metagenomic analysis of bioaerosols to monitor air quality, employing the latest advancements in nanopore sequencing technology. Our workflow is designed to extract, process, and analyze genetic material from air samples to understand microbial compositions. Key stages include:

### Processing of Reads
Initial processing involves quality control and preparation of raw sequencing reads, including error correction and adapter trimming, to ensure data accuracy.

### Base Calling with Guppy and Dorado
Base calling translates the electrical signals from nanopore sequencing into nucleotide sequences using Guppy and Dorado. This critical step converts raw data at a sequencing frequency of 5 kilohertz, enabling precise identification of nucleotides from the electrical signal generated by the DNA passing through nanopores.

### Assembly of Metagenomes
We assemble metagenomes from high-quality reads, reconstructing the genomic sequences of organisms present in bioaerosol samples. This process creates comprehensive genomic profiles for in-depth analysis.

### Binning
Sequences are sorted into bins based on genomic content, separating different organisms' genomes for targeted analysis.

### Quality Control
Quality control is integral at all stages, ensuring the integrity and reliability of data from sequencing to assembly and bining.

### Taxonomic Classification
Our analysis includes taxonomic classification at the read, contig, and metagenome-assembled genome (MAG) levels, providing insights into the diversity and structure of microbial communities.

### Protein Classification and Functional Annotation
We classify proteins and annotate genes and genomes to understand the biological functions and potential environmental impacts of identified organisms.

By integrating high-frequency nanopore sequencing with detailed bioinformatic analysis, this project aims to enhance our understanding of air microbial compositions.

# Air_Metagenomics

![alt text](Workflow.png)

# Nanopore Sequencing Data Analysis Pipeline

This repository contains a pipeline for the analysis of Oxford Nanopore sequencing data. The pipeline uses the following tools:

1. **Guppy Basecaller**: This software tool, provided by Oxford Nanopore Technologies (ONT), is used to convert the raw electrical signal data from nanopore sequencing into DNA sequences, a process known as basecalling.

2. **Porechop**: This tool is developed for Oxford Nanopore sequencing data. It finds and removes adapters from Oxford Nanopore reads.

3. **NanoFilt**: This is a simple tool to filter Oxford Nanopore sequencing data. It reads in a FASTQ file (or stdin), filters reads based on a minimum quality and/or a minimum length, and writes out the filtered reads to stdout.

4. **Flye**: This is a de novo assembler for single-molecule sequencing reads, such as those produced by PacBio and Oxford Nanopore Technologies. It can be run in 'meta' mode for metagenomic projects.

5. **Minimap2**: This tool serves as an aligner for bio-sequences. In the context of nanopore sequencing, it aligns the raw nanopore reads to the assembled sequences produced by Flye. The `-ax map-ont` option specifies that the input data is Oxford Nanopore reads. The output is a BAM file sorted by samtools.

6. **Racon**: It is a tool that performs consensus sequencing. It takes as input the raw nanopore reads, and the alignment file produced by Minimap2, and the assembly produced by Flye, and then it generates a consensus sequence that represents an 'improved' version of the initial assembly. This step is also known as 'polishing' the assembly.

8. **MaxBin2**: MaxBin is an automated binning program to recover individual genomes from metagenomes using an Expectation-Maximization algorithm. This tool groups contigs into individual species based on nucleotide composition and read coverage.

9. **MetaBat2**: MetaBAT is an efficient tool designed to automatically group large genomic fragments assembled from metagenomic shot-gun sequencing. MetaBAT uses a variety of features to group contigs into bins representing individual species.

10. **DAS Tool**: DAS Tool is an automated method that integrates the results of a flexible number of binning algorithms to calculate an optimized, non-redundant set of bins from a single assembly. In this pipeline, it takes the binning results from Vamb, MaxBin2, and MetaBat2 to generate a consolidated and optimized set of bins.

11. **CoverM**: CoverM aims to be a configurable, easy to install, and computationally efficient tool to calculate coverage from mapped metagenomic reads to genomes. It calculates the coverage of the contigs, which can be used to estimate the relative abundance of the species in the sample.

12. **CheckM**: CheckM provides a set of tools for assessing the quality of genomes recovered from isolates, single cells, or metagenomes. It uses lineage-specific marker sets to assess genome completeness and contamination. It provides robust estimates of genome completeness and contamination by using collocated sets of genes that are ubiquitous and single-copy within a lineage.

13. **Quast**: Quast (QUality ASsessment Tool) evaluates and reports on the quality of genome assemblies. It provides statistics such as the number of contigs, the N50 statistic, GC content, and more. The input to Quast is the binned contigs from Vamb.

14. **Kraken 2**: This is a system for assigning taxonomic labels to short DNA sequences. It's generally used in metagenomics projects to identify the species present in a sample. The input to Kraken 2 is the binned contigs from Vamb.

15. **Prodigal**: Prodigal (PROkaryotic DYnamic programming Gene-finding ALgorithm) is a microbial gene prediction program. It predicts protein-coding genes in the contigs. The input to Prodigal is the binned contigs from Vamb.

16. **DIAMOND**: Diamond (Double-Index Alignment of Next-Generation Sequencing Data) is a high-performance bioinformatics software for rapid alignment of DNA sequences to protein databases. It is particularly useful for large-scale genomic analysis, such as metagenomic studies. Diamond operates by translating nucleotide queries into protein sequences and aligning them against a protein reference database, offering speeds significantly faster than traditional BLAST. The typical input for Diamond includes short DNA sequences, often from metagenomic or other genomic datasets.

17. **MetaWRAP**: MetaWRAP is a comprehensive pipeline designed for the efficient analysis of metagenomic data. It integrates several bioinformatics tools and workflows to streamline the process of quality control, assembly, binning, and annotation of metagenomic samples. MetaWRAP simplifies the complex task of analyzing metagenomic data, making it more accessible while maintaining high standards of robustness and accuracy. The input for MetaWRAP typically consists of raw sequencing reads from microbial communities, which are then processed through its various modules to yield meaningful insights into the microbial composition and function.





